### DISCUSSION TITLE
Thumbnail Jobs - if 4x is normal... what about 5-6x?

### DISCUSSION THREAD

**Discussion by streetcap** [immich-app/immich#5485] [Q&A]
Hello,

I realize that the number of thumbnails being processed should be a multiple of how many images you actually have.  This is because, as explained at https://github.com/immich-app/immich/discussions/4869,:

There are multiple thumbnail jobs within that queue.

jpeg thumbnail
webp thumbnail
face thumbnail
thumbhash thumbnail

So should I expect no more than 4x?

Under Server Stats, I have 203,763 photos and 12,111 videos.  My "Generate Thumbnails" is currently at 1,111,736.  So I am clearly above a 4x multiple (I have also been processing thumbnails for many days on top of that).

The reason I ask is because I also noticed someone allegedly "solving" the "problem" by clearing the thumbnail job and reprocessing it:

https://www.reddit.com/r/immich/comments/15eacdc/more_pictures_waiting_than_actual_files/

So I am just looking for clarification if this is a bug or normal behavior.

FYI, it is a completely fresh install that is being indexed (so there shouldn't be any old background jobs still trying to catch up adding to the tally).  After the fresh install I also paused all of the ML jobs and the Transcoding to make sure that my computer has plenty of resources dedicated to the Library, Metadata, and Thumbnail jobs.

Any insight anybody has would be greatly appreciated :)

**Comment by bo0tzz** [comment]
A thumbnail is also generated for every detected face, so if you have a lot of (group) pictures of people then it's conceivable that you might hit this long of a queue.

**Reply by streetcap** [reply]
Thanks for replying :)

Unfortunately, I have had the ML paused.  So, I assume that it hasn't detected any faces.  I can also confirm there are no faces displayed in the "Explore" tab.

Perhaps "Sharing" images between user accounts causes additional thumbnails to be generated??

**Comment by streetcap** [comment]
Just in case this helps anyone else... it appears to be a normal behavior.  My thumbnail jobs capped out at about 6x the number of items in my library.  I suspect it was because of the "Sharing" being between users.  It sounds like it could have been even worse if I also had machine learning enabled as well.
